pkg/tasks/runner.go

```diff
@@ -33,25 +34,78 @@ func RunAll(cfg *config.Config, proj *project.Project, ui *termui.UI) (success b
 	}
 
 	for _, task := range taskList {
-		ctx.ui.TaskHeader(task.name(), task.header())
+		if t, ok := task.(taskWithPreRunValidation); ok {
+			err = t.preRunValidation(ctx)
+			if err != nil {
+				ctx.ui.TaskError(err)
+				return false, nil
```

Comment:
The caller of `RunAll` will display a returned error with a generic (last recourse) template.
It is used above, for `GetTasksFromProject` for instance.

For the errors produced by tasks, we want to keep the control over the error handling (we could even want to ignore it a skip to the next task, in some situation).

In this case we don't return the error since we displayed it properly, but we also signal that the command didn't succeed.

Tell me if it makes sense.

Reply:
I think it does make sense for now, but since a runner is gonna ran one time, will be `fulfilled` or not. It might be an idea in a following to store those errors into the runner itself. Like that we would know after the fact what failed to run.

what do you think?